Sources/Honeycomb/HoneycombNavigationSpanInstrumentation.swift

```diff
@@ -16,76 +21,177 @@ func getTracer() -> Tracer {
 internal class HoneycombNavigationProcessor {
     static let shared = HoneycombNavigationProcessor()
     var currentNavigationPath: [String] = []
+    var lastNavigationTime: Date? = nil
 
-    private init() {}
+    private init() {
+        setupAppLifecycleTracking()
+    }
 
     @available(iOS 16.0, macOS 13.0, *)
-    func reportNavigation(path: NavigationPath) {
+    func reportNavigation(prefix: String? = nil, path: NavigationPath, reason: String? = nil) {
         if let codablePath = path.codable {
-            reportNavigation(path: codablePath)
+            reportNavigation(path: codablePath, reason: reason)
         } else {
-            reportNavigation(path: unencodablePath)
+            reportNavigation(path: unencodablePath, reason: reason)
         }
     }
 
-    func reportNavigation(path: String) {
-        currentNavigationPath = [path]
-
-        // emit a span that says we've navigated to this path
-        getTracer().spanBuilder(spanName: navigationSpanName)
-            .setAttribute(key: "screen.name", value: path)
-            .startSpan()
-            .end()
+    func reportNavigation(prefix: String? = nil, path: String, reason: String? = nil) {
+        reportNavigation(path: [path], reason: reason)
     }
 
-    func reportNavigation(path: Encodable) {
+    func reportNavigation(prefix: String? = nil, path: Encodable, reason: String? = nil) {
         do {
             let encoder = JSONEncoder()
             let data = try encoder.encode(path)
             let pathStr = String(decoding: data, as: UTF8.self)
 
-            reportNavigation(path: pathStr)
+            reportNavigation(path: pathStr, reason: reason)
         } catch {
-            reportNavigation(path: unencodablePath)
+            reportNavigation(path: unencodablePath, reason: reason)
+        }
+    }
+
+    func reportNavigation(prefix: String? = nil, path: [String], reason: String? = nil) {
```

Comment:
it's a little subtle to spot in the git diff but this is an actual functional change here.

Previously, in the version of this function that accepted a `path: [Encodable]`, we would stringify that entire path into a _single_ string. This meant that if that path was empty, our navigation instrumentation would receive `"[]"` (as a string!) as the path. 

Now, that function turns the `[Encodable]` into a `[String]` and calls this function, which calls `navigationStart()`. That function looks at the path and uses the last item in the array as `screen.name`, and uses the `/`-separated version of the array as the `screen.path` attribute.